proposal/random-with-units.md

```diff
@@ -0,0 +1,101 @@
+# Random: Draft 1
+
+*([Issue](https://github.com/sass/sass/issues/1890))*
+
+This proposal defines the behavior of the built-in [`random()`][random]
+function.
+
+[random]: ../spec/built-in-modules/math.md#random
+
+## Table of Contents
+
+* [Background](#background)
+* [Summary](#summary)
+  * [Design Decisions](#design-decisions)
+* [Semantics](#semantics)
+* [Deprecation Process](#deprecation-process)
+
+## Background
+
+> This section is non-normative.
+
+Sass contains a built-in `random()` function which is [documented on the public
+site][random public], but the behavior itself is **not** [specified][].
+
+Currently the function parameter `$limit` (defaults to `null`) changes the
+behavior from returning a decimal in range `[0, 1]` to returning an integer in
+range `[1, $limit]`. An error is thrown when passing a non-numeric value.
+
+It is undefined what happens when the `$limit` parameter contains a unit (e.g.
+`5px` or `8em`), and the current behavior is unexpected as `random()` will
+[drop the units and percent symbol][issue].
+For example: `random(42px) => 28` (there is no `px`).
+
+[random public]: https://sass-lang.com/documentation/modules/math#random
+[specified]: ../spec/built-in-modules/math.md#random
+[issue]: https://github.com/sass/sass/issues/1890
```

Comment:
The background and summary should be written more for the audience of Sass authors than people who are working on the spec. For this audience, it doesn't really matter what's in the spec (and in fact, it can generally be assumed that most aspects of the language _aren't_ formally specified yet). Instead, try focusing your explanation on what the `random()` function actually does in practice, and why that's undesirable.